### PR TITLE
Migrate documentation to readthedocs.io

### PR DESCRIPTION
**Pull request by ddworken**
This PR migrates all of the documentation to readthedocs.io instead of having a giant README.md file. The docs now live in `docs/` as markdown files that are rendered to HTML via sphinx using the readthedocs theme. 

Running `make` inside of the `docs/` folder will output help info about using sphinx to render to HTML locally. But generally speaking, it isn't necessary to do anything when editing docs. Just updating a markdown file and merging to master will trigger readthedocs to update. 

See the readthedocs website for the output of this:

https://keybase-ssh-ca-bot.readthedocs.io/en/latest/

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Minimal makefile for Sphinx documentation
```
Comment (ddworken) [inline docs/Makefile]: Default makefile from sphinx

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Architecture
```
Comment (ddworken) [inline docs/architecture.md]: These are new docs

File: docs/best_practices.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Best Practices
```
Comment (ddworken) [inline docs/best_practices.md]: These are new docs

File: docs/conf.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
```
Comment (ddworken) [inline docs/conf.py]: Default sphinx config file for readthedocs.io

File: docs/index.rst [hunk 5]
```diff
@@ -0,0 +1,16 @@
+Keybase SSH CA Bot Documentation!
+==============================================
+
+.. toctree::
+   :maxdepth: 2
+
+   intro
```
Comment (ddworken) [inline docs/index.rst]: This defines the list of markdown files to render in what order

File: .gitignore [hunk 6]
```diff
@@ -3,4 +3,9 @@ keybaseca.config
 nohup.out
 env.sh
 __pycache__
-generated-env
\ No newline at end of file
+generated-env
```
Comment (xgess) [inline .gitignore]: i think this is old now, no? looks like it's only in the tests directory.

File: docs/architecture.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+![Architecture Diagram](https://raw.githubusercontent.com/keybase/bot-sshca/master/docs/Architecture%20Diagram.png "Architecture Diagram")
+
+Note that this means that you do not need to modify your servers in any
+way or run any additional processes on your servers other than a standard 
+OpenSSH daemon. 
+
+## Network Architecture
+
+Since all communication between the kssh client and the SSH CA server happens over Keybase chat, it is possible (and recommended)
+to firewall off the SSH CA server so it cannot be reached from the general internet. Additionally, note that the SSH servers
```
Comment (xgess) [inline docs/architecture.md]:
```suggestion
to firewall off the SSH CA server (where this bot is running) so it cannot be reached from the general internet. Additionally, note that the SSH servers
```

Comment (xgess) [reply docs/architecture.md]: there are a few places we mention keybaseca and sshca server, and it's probably worth periodically hammering home that this is the bot.

File: docs/best_practices.md (hunk 3 above)
Comment (xgess) [reply docs/best_practices.md]: love the idea of a best practices section

File: .gitignore (hunk 6 above)
Comment (ddworken) [reply .gitignore]: Yup, that directory no longer exists or is needed. Removed from the gitignore (if you still have it, just blow it away).

File: docs/best_practices.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+you get. 
+
+If you are using other bots in the same teams as the SSH CA bot (or if you wish to have normal conversation in those 
+teams), you can use the `CHAT_CHANNEL` environment variable in order to configure a specific chat channel for all
+SSH CA messages. 
+
+## Network Isolation
+
+Due to the highly sensitive nature of the SSH CA bot, it is recommended to configure firewalls in order to block all
+access to the server running the CA bot. It is not recommended to use the CA system to access the CA bot itself
```
Comment (xgess) [inline docs/best_practices.md]:
```suggestion
access to the server running the CA bot. It is not recommended to use kssh to access the server of the CA bot itself
```

File: docs/contributing.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+Prior to sending a `SignatureRequest`, kssh sends a series of `AckRequest` messages. An `AckRequest` message is sent until 
+kssh receives an `Ack` from keybaseca. This is done in order to ensure that kssh has correctly connected to the chat channel
+and that the bot is responding to messages. Afterwards, a `SignatureRequest` packet is sent and keybaseca parses it and 
+returns a signed key. Note that only public keys and signatures are sent over Keybase chat and private keys never 
+leave the devices they were generated on. 
+
+#### SSH Operations
+
+SSH keys are generated in pure go code. All generated SSH keys are ed25519 private keys. keybaseca uses the ssh-keygen
```
Comment (xgess) [inline docs/contributing.md]: this is no longer strictly true, yes?

File: docs/contributing.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+#### KBFS
+
+In order to ensure that keybaseca can run inside of docker (which does not support FUSE filesystems without adding
+the CAP_SYS_ADMIN permission), all KBFS interactions are done via `keybase fs ...` commands. This makes it so that 
+keybaseca can run in unprivileged docker containers. 
+
+## Integration Tests
+
+This project contains integration tests that can be run via `./integrationTest.sh`. The integration tests depend on 
+docker and docker-compose. The first time you run them, they will walk you through creating two new accounts to be
```
Comment (xgess) [inline docs/contributing.md]:
```suggestion
docker and docker-compose. The first time you run them, they will walk you through creating two new live keybase accounts to be 
```

File: docs/env.md [hunk 11]
```diff
@@ -84,7 +86,7 @@ export CHAT_CHANNEL="team.prod#ssh-provision"
 export CHAT_CHANNEL="team.ssh_bot#general"
 ```
 
-### Developer Options
+## Developer Options
 
 These environment variables are mainly useful for dev work. For security reasons, it is recommended to always run a
```
Comment (xgess) [inline docs/env.md]:
```suggestion
These environment variables are mainly useful for dev work. For security reasons, it is recommended always to run a 
```

File: docs/getting_started.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+Start by creating a new Keybase user to use for the CA chatbot:
+
+```bash
+keybase signup      # Creates a new Keybase user to use for the SSH CA bot
+keybase paperkey    # Generate a new paper key
+```
+
+Note that this system will not work if you attempt to use the same user for the CA chatbot as for kssh. It is required
+to use a unique user.
```
Comment (xgess) [inline docs/getting_started.md]:
```suggestion
to use distinct users. 
```

File: docs/getting_started.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+Note that this system will not work if you attempt to use the same user for the CA chatbot as for kssh. It is required
+to use a unique user. 
+
+Then create `{TEAM}.ssh.staging`, `{TEAM}.ssh.production`, `{TEAM}.ssh.root_everywhere` as new Keybase subteams
+and add the bot to those subteams. Add users to those subteams based off of the permissions you wish to grant
+different users
+
+On a secured server that you wish to use to run the CA chatbot:
```
Comment (xgess) [inline docs/getting_started.md]: the only thing that needs to be on this server is docker, yes? that might be worth mentioning right here.

File: docs/getting_started.md [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+cp env.sh.example env.sh
+nano env.sh         # Fill in the values including the previously generated paper key
+make generate       # Generate a new CA key
+```
+
+Running `make generate` will output a list of configuration steps to run on each server you wish to use with the CA chatbot. 
+These commands create a new user to use with kssh (the `developer` user), add the CA's public key to the server, and 
+configure the server to trust the public key. 
+
+Now you must define a mapping between Keybase teams the users and servers that they are
```
Comment (xgess) [inline docs/getting_started.md]:
```suggestion
Now you must define a mapping between Keybase teams and the users on the servers that they are
```

File: docs/getting_started.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+allowed to access. If you wish to make the user foo available to anyone in team.ssh.bar,
```
Comment (xgess) [inline docs/getting_started.md]:
```suggestion
allowed to access. If you wish to make the user `foo` on your server available to anyone in team.ssh.bar,
```

File: docs/os_support.md [hunk 16]
```diff
@@ -0,0 +1,8 @@
+# OS Support
+
+It is recommended to run the server component of this bot on linux and running it in other environments is untested.
+`kssh` is tested and works correctly on linux, macOS, and Windows. If running on windows, note that there is a dependency
```
Comment (xgess) [inline docs/os_support.md]:
```suggestion
`kssh` is tested and works correctly on Linux, macOS, and Windows. If running on Windows, note that there is a dependency
```

File: docs/troubleshooting.md [hunk 17]
```diff
@@ -25,7 +25,9 @@ It means that for whatever reason, kssh is not receiving a response from the CA
 Keybase chat. First, ensure that the CA chatbot is currently running. Next, attempt to determine what is happening
 by inspecting the chat messages inside of the teams configured with the chatbot. You should see a series of `Ack` and 
 `AckRequest` messages going back and forth prior to a `Signature_Request:` and a `Signature_Response:` exchange. Ensure 
-that you and the chatbot are in the correct teams such that they can read and respond to the messages. 
+that you and the chatbot are in the correct teams such that they can read and respond to the messages. In addition,
+review the log output from the keybaseca chatbot. Note that it is required to run the keybaseca chatbot as a different
+user than you are using for kssh.
```
Comment (xgess) [inline docs/troubleshooting.md]:
```
Note that it is required to run the keybaseca chatbot as a different user than you are using for kssh. 
```
i like that this is called out multiple times. guaranteed people will still make this mistake, but it's such an obvious mistake, it's good to try to preempt.

File: docs/contributing.md (hunk 9 above)
Comment (ddworken) [reply docs/contributing.md]: Updated

File: docs/getting_started.md (hunk 13 above)
Comment (ddworken) [reply docs/getting_started.md]: Yup! Added a comment

**Pull request merged**